scripts/check_site_lib.lua

```diff
@@ -133,8 +133,9 @@ function need_one_of(varname, array, required)
 end
 
 function need_string_array(varname, required)
-   return assert(pcall(need_array, varname, function(e) assert_type(e, 'string') end, required),
-		 "site.conf error: expected `" .. varname .. "' to be a string array")
+   assert(pcall(need_array, varname, function(e) assert_type(e, 'string') end, required),
```

Comment:
Hmm, the `pcall` should return two values, a true/false value and the original return value of `need_array`, we're just discarding the second one by passing it to assert. I think we could do without the extra `loadvar` like this (untested):

```
local ok, var = pcall(need_array, varname, function(e) assert_type(e, 'string') end, required)
assert(ok, "site.conf error: expected `" .. varname .. "' to be a string array")
return var
```